scripts/git-clang-format

```diff
@@ -0,0 +1,660 @@
+#!/usr/bin/env python3
```

Comment:
Is this a copy of the git-clang-format file that can be installed via 'dnf install git-clang-format'?

I would avoid creating our own copy that we'd have to maintain if possible and require folks to install this via their distro packages.

If their distro does not have a git-clang-format package then they can download it from llvm project as described in [code-conventions.md](https://github.com/skupperproject/skupper-router/blob/ed76beacbefac284df34400760df00bbc2b25dee/docs/notes/code-conventions.md?plain=1#L26)

Reply:
> Is this a copy of the git-clang-format file that can be installed via 'dnf install git-clang-format'?

Yes

> I would avoid creating our own copy that we'd have to maintain if possible and require folks to install this via their distro packages.

It's going to be low maintenance, and it is going to go away when all .c/.h files are correctly formatted; then we would no longer need to use it to limit the check to the changed lines.

I'm going to leave it in the PR for now, not touch anything, until you make the precommit work on your machine. Then we may make changes.

Reply:
Sounds good, thanks

Reply:
I'm going to write something to README.adoc, to explain things

Reply:
+1 README.adoc.